### PR TITLE
scheduler: cache resource limit lookup in tick

### PR DESCRIPTION
**Pull request by danielnorberg**
Alternative to #555

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java [hunk 1]
```diff
@@ -221,13 +223,15 @@ private void dequeueInstances(
       Map<WorkflowId, Workflow> workflows,
       List<InstanceState> eligibleInstances) {
 
+    final ConcurrentMap<String, Boolean> resourceExhaustedCache = new ConcurrentHashMap<>();
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java]: Where do we invalidate the cache?

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java]: A new instance is created for for every tick.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java]: Ah ok i read it as instance variable.

**Comment by danielnorberg** [comment]
@honnix Still reviewing this?

**Comment by honnix** [comment]
+1

**Pull request merged**